presto-spi/src/main/java/com/facebook/presto/spi/InMemoryRecordSet.java

```diff
@@ -234,6 +235,9 @@ else if (VARCHAR.equals(type)) {
                     checkArgument(value instanceof String || value instanceof byte[],
                             "Expected value %d to be an instance of String or byte[], but is a %s", i, value.getClass().getSimpleName());
                 }
+                else if (TIMESTAMP_WITH_TIME_ZONE.equals(type)) {
```

Comment:
Merge this with `else if (BIGINT.equals(type) || DATE.equals(type) || TIMESTAMP.equals(type)) {`